src/main/java/com/unstoppabledomains/resolution/naming/service/ZNS.java

```diff
@@ -68,6 +68,37 @@ public String getOwner(String domain) throws NamingServiceException {
         return addresses[0];
     }
 
+    @Override
+    public List<String> batchOwners(List<String> domains) throws NamingServiceException {
```

Comment:
Same as for ENS implementation

Reply:
Should I throw a not implemented error instead of this implementation? I feel that a slow implementation is better than no implementation at all.

Reply:
@JohnnyJumper Please throw an error for now.